libobs/obs-source.c

```diff
@@ -339,6 +345,26 @@ static void obs_source_init_audio_hotkeys(struct obs_source *source)
 		obs_source_hotkey_push_to_talk, source);
 }
 
+static void obs_source_deduplicate_name(const char **name)
+{
+	struct obs_context_data *tmp;
+	HASH_FIND_STR((struct obs_context_data *)obs->data.sources, *name, tmp);
+
+	if (tmp) {
+		struct dstr new_name = {0};
+		int suffix = 2;
+
+		while (tmp) {
+			dstr_free(&new_name);
```

Comment:
I don't think it's necessary to call `dstr_free` here.

Reply:
Yeah it probably isn't since `dstr_printf` will just overwrite or reallocate the memory, I'll remove it.